core/src/apps/cardano/sign_tx.py

```diff
@@ -316,7 +298,39 @@ def _validate_metadata(metadata: Optional[bytes]) -> None:
         raise INVALID_METADATA
 
 
-def _serialize_tx(keychain: seed.Keychain, msg: CardanoSignTx) -> Tuple[bytes, bytes]:
+def _get_signed_tx_chunks(
+    cborized_tx: CborizedSignedTx,
+) -> Iterator[Tuple[bytes, bool]]:
+    remaining_len = 0
+    for token in cbor.encode_chunked(cborized_tx):
+        remaining_len += len(token)
+
+    chunks = cbor.encode_chunked(cborized_tx)
+    current_chunk_view = utils.BufferReader(b"")
+    while remaining_len > 0:
+        payload_len = min(remaining_len, MAX_TX_CHUNK_SIZE)
+        payload_writer = utils.BufferWriter(bytearray(payload_len))
```

Comment:
you could also rewrite this to reuse the same `bytearray` again and again; see what `bitcoin.py` does with `self.serialized_tx`, and how `bitcoin/sign_tx/helpers.py` send out and clear its contents.

feel free to leave this optimization to a followup PR though

Reply:
wouldn't there be an issue with sending out the last chunk of the tx which may not fix perfectly into the bytearray? Or would you suggest sending back the fixed size bytearray together alongside its length?

Reply:
from the bitcoin helpers.py it seems to me as if bitcoin was flushing that pre-allocated buffer with 2048 bytes on each interaction with the client, but I can't quite see where do the trailing null bytes in the pre-allocated buffer get sliced/omitted from the protobuf message itself - I'm probably missing something very basic there

Reply:
size of bytearray contents can be smaller than the allocated capacity
```
tx_req.serialized.serialized_tx[:] = bytes()
```
is resetting the bytearray to a zero-size, but no reallocation happens and the backing buffer is still at 2048 bytes

Reply:
see also the definition of `empty_bytearray` at `writers.py`